File: java/socketd/src/main/java/org/noear/socketd/transport/core/Stream.java
package org.noear.socketd.transport.core;

import org.noear.socketd.utils.IoConsumer;

import java.util.function.BiConsumer;
import java.util.function.Consumer;

/**
 * 流接口
 *
 * @author noear
 * @since 2.1
 */
public interface Stream {
    /**
     * 流Id
     */
    String sid();

    /**
     * 是否单收
     */
    boolean isSingle();

    /**
     * 是否完成
     */
    boolean isDone();

    /**
     * 超时设定（单位：毫秒）
     */
    long timeout();

    /**
     * 异常发生时
     */
    Stream thenError(Consumer<Throwable> onError);

    /**
     * 进度发生时
     */
    Stream thenProgress(BiConsumer<Integer, Integer> onProgress);
}

File: swift/socketd/socketd/src/socketd/transport/core/Stream.swift
//
//  Stream.swift
//  socketd
//
//  Created by noear on 2023/12/27.
//

import Foundation

/**
 * 流接口
 *
 * @author noear
 * @since 2.1
 */
protocol Stream{
    /**
     * 流Id
     */
    func sid() -> String;
    
    /**
     * 是否单收
     */
    func isSingle() -> Bool;
    
    /**
     * 是否完成
     */
    func isDone() -> Bool;
    
    /**
     * 超时设定（单位：毫秒）
     */
    func timeout() -> Int64;
    
    /**
     * 异常发生时
     */
    func thenError(_error:Error) -> Stream;
}
